ethers-etherscan/src/account.rs

```diff
@@ -0,0 +1,692 @@
+use std::{
+    borrow::Cow,
+    collections::HashMap,
+    fmt::{Display, Error, Formatter},
+};
+
+use ethers_core::{
+    abi::Address,
+    types::{BlockNumber, Bytes, H256, U256, U64},
+};
+use serde::{Deserialize, Serialize};
+
+use crate::{Client, EtherscanError, Query, Response, Result};
+
+/// The raw response from the balance-related API endpoints
+#[derive(Debug, Serialize, Deserialize)]
+pub struct AccountBalance {
+    pub account: Address,
+    pub balance: String,
+}
+
+mod jsonstring {
+    use super::*;
+    use serde::{
+        de::{DeserializeOwned, Error as _},
+        ser::Error as _,
+        Deserializer, Serializer,
+    };
+
+    pub fn serialize<T, S>(
+        value: &GenesisOption<T>,
+        serializer: S,
+    ) -> std::result::Result<S::Ok, S::Error>
+    where
+        T: Serialize,
+        S: Serializer,
+    {
+        let json = match value {
+            GenesisOption::None => Cow::from(""),
+            GenesisOption::Genesis => Cow::from("GENESIS"),
+            GenesisOption::Some(value) => {
+                serde_json::to_string(value).map_err(S::Error::custom)?.into()
+            }
+        };
+        serializer.serialize_str(&json)
+    }
+
+    pub fn deserialize<'de, T, D>(
+        deserializer: D,
+    ) -> std::result::Result<GenesisOption<T>, D::Error>
+    where
+        T: DeserializeOwned,
+        D: Deserializer<'de>,
+    {
+        let json = Cow::<'de, str>::deserialize(deserializer)?;
+        if !json.is_empty() && !json.starts_with("GENESIS") {
+            let value =
+                serde_json::from_str(&format!("\"{}\"", &json)).map_err(D::Error::custom)?;
+            Ok(GenesisOption::Some(value))
+        } else if json.starts_with("GENESIS") {
+            Ok(GenesisOption::Genesis)
+        } else {
+            Ok(GenesisOption::None)
+        }
+    }
+}
+
+/// Possible values for some field responses
+#[derive(Debug)]
+pub enum GenesisOption<T> {
+    None,
+    Genesis,
+    Some(T),
+}
```

Comment:
we should add helper functions here to
`is_genesis`
`value(&self) -> Option<&T>`